test/SafeProtocolRegistry.spec.ts

```diff
@@ -85,4 +91,28 @@ describe("SafeProtocolRegistry", async () => {
         expect(await safeProtocolRegistry.supportsInterface.staticCall("0xbaddad42")).to.be.false;
         expect(await safeProtocolRegistry.supportsInterface.staticCall("0xffffffff")).to.be.false;
     });
+
+    it("Should revert when adding hooks not supporting expected interfaceId", async () => {
+        const { safeProtocolRegistry } = await loadFixture(deployContractFixture);
+        const mockHookAddress = await (await getHooksWithFailingCallToSupportsInterfaceMethod()).getAddress();
+        await expect(safeProtocolRegistry.connect(owner).addIntegration(mockHookAddress, IntegrationType.Hooks))
+            .to.be.revertedWithCustomError(safeProtocolRegistry, "IntegrationDoesNotSupportExpectedInterfaceId")
+            .withArgs(mockHookAddress, "0x907e1c56");
+    });
+
+    it("Should revert when adding plugin not supporting expected interfaceId", async () => {
+        const { safeProtocolRegistry } = await loadFixture(deployContractFixture);
+        const mockPluginAddress = await (await getPluginWithFailingCallToSupportsInterfaceMethod()).getAddress();
+        await expect(safeProtocolRegistry.connect(owner).addIntegration(mockPluginAddress, IntegrationType.Plugin))
+            .to.be.revertedWithCustomError(safeProtocolRegistry, "IntegrationDoesNotSupportExpectedInterfaceId")
+            .withArgs(mockPluginAddress, "0x3fce835e");
+    });
+
+    it("Should revert when adding function handler not supporting expected interfaceId", async () => {
+        const { safeProtocolRegistry } = await loadFixture(deployContractFixture);
+        const mockFunctionHandlerAddress = await (await getFucntionHandlerWithFailingCallToSupportsInterfaceMethod()).getAddress();
```

Comment:
i'd put this into the `deployContractFixture` function. I'd also use `deployContractFixture` for setting up tests boilerplate like it's done in the safe contracts repo